pyscf_ipu/experimental/primitive.py

```diff
@@ -14,6 +15,7 @@ class Primitive:
     alpha: float = 1.0
     lmn: Int3 = jnp.zeros(3, dtype=jnp.int32)
     norm: Optional[float] = None
+    atom_index: Optional[int] = None
```

Comment:
This is a back-pointer into the list of which this is a member?
That's also a hint that the `where(a.atom_index == atom_index)` could be lifted.